mastermind_spec.rb

```diff
@@ -1,40 +1,63 @@
 require_relative './mastermind'
 
 RSpec.describe Mastermind do
+  game = Mastermind.new
   it "prints game instructions" do
-    game = Mastermind.new
-    expect {game.print_instructions}.to output(/Welcome to Mastermind!/).to_stdout
+    expect {game.print_instructions}.to output(/After each guess/).to_stdout
   end
 
+  it "counts the number of each color in a given code" do
+    expect(game.color_count(["R", "O", "Y", "G"])).to eq ({:R => 1, :O => 1, :Y => 1, :G => 1})
+    expect(game.color_count(["P", "P", "P", "P"])).to eq ({:P => 4})
+    expect(game.color_count(["R", "O", "B", "R", "B", "B"])).to eq ({:R => 2, :O => 1, :B => 3})
+  end
+
+  # it "counts how many colors in the guess match the code" do
+  #   expect(game.compare(["R", "O", "Y", "G"], "RROO")).to eq (2)
+  # end
+end
+
+RSpec.describe SecretCode do
+  secret_code = SecretCode.new
```

Comment:
let(:secret_code) { SecretCode.new }. Let bindings are for variables used in tests

Reply:
Awesome, thank you!